File: app/src/main/java/com/davismiyashiro/expenses/view/opentab/ParticipantInterfaces.java
package com.davismiyashiro.expenses.view.opentab;

import com.davismiyashiro.expenses.datatypes.Participant;
import com.davismiyashiro.expenses.datatypes.Tab;

import java.util.List;

/**
 * Created by Davis Miyashiro on 09/12/2016.
 */

public interface ParticipantInterfaces {
    interface ParticipantView {
        void showParticipants (List<Participant> participants);
    }

    interface UserActionsListener {

        void setParticipantView (ParticipantView view);

        void loadParticipants (Tab tab);

        void removeParticipant (Participant participant);

    }
}

File: app/src/main/java/com/davismiyashiro/expenses/view/opentab/ParticipantInterfaces.kt
package com.davismiyashiro.expenses.view.opentab

import com.davismiyashiro.expenses.datatypes.Participant
import com.davismiyashiro.expenses.datatypes.Tab

/**
 * Created by Davis Miyashiro on 09/12/2016.
 */

interface ParticipantInterfaces {
    interface ParticipantView {
        fun showParticipants(participants: MutableList<Participant>)
    }

    interface UserActionsListener {

        fun setParticipantView(view: ParticipantView)

        fun loadParticipants(tab: Tab)

        fun removeParticipant(participant: Participant)

    }
}
